File: app/src/main/java/opencrowd/hgc/hgcwallet/modals/PublicKeyAddress.java
package opencrowd.hgc.hgcwallet.modals;


import org.spongycastle.util.encoders.Hex;

public class PublicKeyAddress {
    String hexAddress;
    private PublicKeyAddress(String hexAddress) {
        this.hexAddress = hexAddress;
    }
    public static PublicKeyAddress from(String string) {
        try {
            if (string == null || string.isEmpty() || Hex.decode(string) == null) return null;
            return new PublicKeyAddress(string);
        } catch (Exception e) {
            return null;
        }
    }

    public String stringRepresentation() {
        return hexAddress;
    }
}

File: app/src/main/java/hedera/hgc/hgcwallet/modals/PublicKeyAddress.kt
package hedera.hgc.hgcwallet.modals

import org.spongycastle.util.encoders.Hex

class PublicKeyAddress private constructor(val hexAddress: String) {

    fun stringRepresentation(): String {
        return hexAddress
    }

    fun getByteArray(): ByteArray {
        return Hex.decode(hexAddress)
    }

    companion object {
        val lengthED = 32
        fun from(str: String): PublicKeyAddress? {
            return try {
                if (str.isBlank() ||  Hex.decode(str).let { it == null || it.size != lengthED }) null else PublicKeyAddress(str)
            } catch (e: Exception) {
                null
            }
        }
    }
}